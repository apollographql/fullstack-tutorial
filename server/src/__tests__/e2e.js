// import our production apollo-server instance
const { server } = require('../');
const gql = require('graphql-tag');

const { startTestServer, toPromise } = require('./__utils');

const LAUNCH_LIST_QUERY = gql`
  query myLaunches($pageSize: Int, $after: String) {
    launches(pageSize: $pageSize, after: $after) {
      cursor
      launches {
        mission {
          name
          missionPatch
        }
      }
    }
  }
`;

const GET_LAUNCH = gql`
  query launch($id: ID!) {
    launch(id: $id) {
      id
      isBooked
      rocket {
        type
      }
      mission {
        name
      }
    }
  }
`;

describe('Server - e2e', () => {
  let stop, graphql;

  beforeEach(async () => {
    const testServer = await startTestServer(server);
    stop = testServer.stop;
    graphql = testServer.graphql;
  });

  afterEach(() => stop());

  it('gets list of launches', async () => {
    const res = await toPromise(
      graphql({
        query: LAUNCH_LIST_QUERY,
        variables: { pageSize: 1, after: '1517949900' },
      }),
    );

    expect(res).toMatchSnapshot();
  });

  it('gets a single launch', async () => {
    const res = await toPromise(
      graphql({ query: GET_LAUNCH, variables: { id: 30 } }),
    );

    expect(res).toMatchSnapshot();
  });
});
