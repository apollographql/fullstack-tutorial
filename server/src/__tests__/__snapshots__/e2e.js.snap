// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e gets a single launch 1`] = `
Object {
  "data": Object {
    "launch": Object {
      "id": "30",
      "isBooked": false,
      "mission": Object {
        "name": "Thaicom 8",
      },
      "rocket": Object {
        "type": "FT",
      },
    },
  },
}
`;

exports[`Server - e2e gets list of launches 1`] = `
Object {
  "data": Object {
    "launches": Object {
      "cursor": "1517433900",
      "launches": Array [
        Object {
          "mission": Object {
            "missionPatch": "https://images2.imgbox.com/42/0a/LAupFe3L_o.png",
            "name": "SES-16 / GovSat-1",
          },
        },
      ],
    },
  },
}
`;
